north/src/runtime/state.rs

```diff
@@ -320,13 +312,88 @@ impl State {
         }
     }
 
+    async fn is_installed(&self, npk: &Path) -> Result<bool, InstallFailure> {
+        let container_id = extract_manifest(npk.into(), &self.signing_keys)?.name;
+        Ok(self.applications.get(&container_id).is_some())
+    }
+
     /// Install a npk from give path
-    // pub async fn install(&mut self, npk: &Path) -> result::Result<(), Error> {
-    //     npk::install(self, npk)
-    //         .await
-    //         .map_err(Error::InstallationError)?;
-    //     Ok(())
-    // }
+    pub async fn install(
+        &mut self,
+        npk: &Path,
+        msg_id: MessageId,
+        container_dir: Option<std::path::PathBuf>,
```

Comment:
We should not use the `std::path` internally - everything is async_std::path`

Reply:
let's do this when porting to tokio